frontend/apps/crates/entry/home/src/home/dom/iframe.rs

```diff
@@ -0,0 +1,49 @@
+use dominator::{clone, html, Dom};
+use dominator_helpers::events::Message;
+use futures_signals::signal::{Mutable, SignalExt};
+use std::rc::Rc;
+use web_sys::HtmlIFrameElement;
+use serde::{Serialize, Deserialize};
+
+const STR_IFRAME_URL: &str = "https://www.jewishinteractive.org/jigzi-home";
+const INT_IFRAME_PADDING: usize = 30;
+const INT_INITIAL_HEIGHT: usize = 3000;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct IframeMessageData {
+    kind: String,
+    height: usize,
+}
+
+pub struct Iframe {
+    height: Mutable<usize>,
+}
+
+impl Iframe {
+    pub fn new() -> Rc<Self> {
+        Rc::new(Self {
+            height: Mutable::new(INT_INITIAL_HEIGHT),
+        })
+    }
+    pub fn render(self: Rc<Self>) -> Dom {
+        let state = self.clone();
+        html!("iframe" => HtmlIFrameElement, {
+            .style("width", "100%")
+            .style("border", "none")
+            .style_signal("height", state.height.signal()
+                .map(|height| {
+                    let adjusted_height = height.to_string() + "px";
+                    adjusted_height
+                })
+            )
+            .global_event(clone!(state => move |event: Message| {
```

Comment:
.event() doesnt work here. I believe this is because we NEED to listen for the message event on the Window, and not the iframe

Reply:
Yeah, looks like you're right